feat: many bubbles

fjebaker opened the pull request:
Renamed character to Player, and the player can now control a variety of bubbles. Each bubble is stored as an entry in a table. Inserting to this table will introduce a new bubble.

Added basic animation so the bubbles oscillate backwards and forwards.

fjebaker committed:
feat: many bubbles

Renamed character to Player, and the player can now control a variety of
bubbles. Each bubble is stored as an entry in a table. Inserting to this
table will introduce a new bubble.

Added basic animation so the bubbles oscillate backwards and forwards.

JR-Mitchell commented on teal/game/player.tl:
```diff
@@ -0,0 +1,120 @@
+require("shared.init")
+
+local DEGREES_TO_RADIANS <const> = 3.1415 / 180
```
You can use `math.pi` rather than 3.1415 if you want

JR-Mitchell commented on teal/game/player.tl:
```diff
@@ -0,0 +1,120 @@
+    end
+    if (keyStates["a"] and not keyStates["d"]) then
+        self.spriteIndex = 0
+
```
Newline?

JR-Mitchell commented on teal/game/player.tl:
```diff
@@ -0,0 +1,120 @@
+    instance.bubbles = {}
+    table.insert(instance.bubbles, Bubble.new())
+    table.insert(instance.bubbles, Bubble.new())
+    instance.bubbles[2].x = 80
```
Maybe we could pass these to the bubble initialiser?
You can do something like
```lua
function Bubble.new(x?: number, y?: number)
    -- existing stuff
    instance.x = x or 30
    instance.y = y or config.displayHeight - 16
    
```

fjebaker replied:
Will do so when it's needed.

fjebaker committed:
feat: multiple bubble sizes

Sprite sheet is set to have 32x32 pixel tiles for the bubbles. There are
now 4 different bubble sizes.

JR-Mitchell reviewed (approved):
:shipit:
